### PR TITLE
[patch] fix PR trigger - (inferBranch flag not work when has default branch)

### PR DESCRIPTION
**Pull request by cjtim**
**What's the change or impact of this PR?**:
* fix: User cannot trigger a pull request using `inferBranch` flag when has default branch set.
* fix: Cannot read MR_IID with the trigger token, switch to Samsahai global service account
* 
**Related issue link**:

### CODE REVIEW

File: internal/staging/util_test.go [hunk 1]
```diff
@@ -42,21 +39,21 @@ var _ = Describe("Util", func() {
 
 			tryInferPullRequestGitlabBranch(gitlabConf, MRiid, gitlabClientGetter)
 
-			g.Expect(gitlabConf).To(Equal(before))
+			g.Expect(gitlabConf).ToNot(Equal(before))
+			g.Expect(gitlabConf.Branch).To(Equal(inferredBranch))
 		})
 
-		It("should infer if true and branch empty", func() {
+		It("interBranch=True defaultBranch='test2'", func() {
```
Comment (Pohfy123) [inline internal/staging/util_test.go:46]: Would it be better to add a description here like what will happen if users set this and that?

File: internal/staging/util_test.go [hunk 2]
```diff
@@ -20,17 +19,15 @@ var _ = Describe("Util", func() {
 				PipelineTriggerToken: "xyz",
 			}
 			MRiid = "87878"
-			gitlabClientGetter = nil
 		})
 
-		It("should not infer if false", func() {
-			gitlabConf.SetInferBranch(false)
+		It("interBranch=True defaultBranch=''", func() {
```
Comment (Pohfy123) [inline internal/staging/util_test.go:24]: Would it be better to add a description here like what will happen if users set this and that?

File: internal/staging/util_test.go [hunk 3]
```diff
@@ -67,20 +64,16 @@ var _ = Describe("Util", func() {
...
 			g.Expect(gitlabConf.Branch).To(Equal(inferredBranch))
-			gitlabConf.Branch = before.Branch
-			// should change only branch field
-			g.Expect(gitlabConf).To(Equal(before))
 		})
 
-		It("should not infer if true and branch not empty", func() {
-			gitlabConf.SetInferBranch(true)
-			gitlabConf.Branch = "abc"
+		It("inferBranch=False defaultBranch=''", func() {
```
Comment (Pohfy123) [inline internal/staging/util_test.go:69]: This one as well?

File: internal/staging/util_test.go [hunk 4]
```diff
@@ -92,6 +85,27 @@ var _ = Describe("Util", func() {
 
 			tryInferPullRequestGitlabBranch(gitlabConf, MRiid, gitlabClientGetter)
 
+			g.Expect(gitlabConf).ToNot(Equal(before))
+			g.Expect(gitlabConf.Branch).To(Equal(inferredBranch))
+		})
+
+		It("inferBranch=False defaultBranch='test4'", func() {
```
Comment (Pohfy123) [inline internal/staging/util_test.go:92]: Also this one

File: internal/staging/util_test.go (hunk 1 above)
Comment (cjtim) [reply internal/staging/util_test.go]: added. Please help review again.

**Review by Pohfy123 — APPROVED** [review]

**Review by sunny299 — APPROVED** [review]

**Pull request merged**